### PR TITLE
[1250] Support for well decommission report 

### PR DESCRIPTION
**Pull request by stephenhillier**
Various types of reports are supported by declaring the form "wizard steps" that must be completed to submit that type of form.

This pull request does not include any of the decommission report specific forms. Those are covered under separate tickets.

Note: commits were taken from @Sybrand 's issues branch- they will disappear from this PR once those commits are merged (so hold off on reviewing those commits in this PR)

### CODE REVIEW

File: app/backend/wells/serializers.py [hunk 1]
```diff
@@ -67,6 +68,22 @@ class Meta:
         }
 
 
+class DecommissionDescriptionSerializer(serializers.ModelSerializer):
+    """Serializes Decommission Descriptions"""
+
+    class Meta:
+        model = DecommissionDescription
+        fields = (
+            'decommission_description_guid',
```
Comment (Sybrand) [inline app/backend/wells/serializers.py]: Technically we don't need this right now; but if we need this in the future depends on if we're going to "stack" de-commission alterations.

File: app/frontend/src/submissions/components/SubmissionForm/ClosureDescription.vue [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  mixins: [inputBindingsMixin],
+  props: {
+    closureDescriptionSet: {
+      type: Array,
+      default: () => ([])
+    },
+    errors: Array,
+    fieldsLoaded: Object
+  },
+  fields: {
```
Comment (Sybrand) [inline app/frontend/src/submissions/components/SubmissionForm/ClosureDescription.vue]: Do you want to keep on using fields?

File: app/frontend/src/submissions/components/SubmissionForm/ClosureDescription.vue [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    getFieldsLoaded (index) {
+      if (this.fieldsLoaded && 'closure_description_set' in this.fieldsLoaded && index in this.fieldsLoaded['closure_description_set']) {
+        return this.fieldsLoaded['closure_description_set'][index]
+      }
+      return {}
+    }
+  },
+  created () {
+    if (!this.closureDescriptionSet.length) {
+      this.addClosureRow()
```
Comment (Sybrand) [inline app/frontend/src/submissions/components/SubmissionForm/ClosureDescription.vue]: Have you checked with Lindsay if she wants three rows?

File: app/frontend/src/submissions/components/SubmissionForm/DecommissionInformation.vue [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    errors: {
+      type: Object,
+      default: () => ({})
+    },
+    fieldsLoaded: {
+      type: Object,
+      default: () => ({})
+    }
+  },
+  fields: {
```
Comment (Sybrand) [inline app/frontend/src/submissions/components/SubmissionForm/DecommissionInformation.vue]: Again, do you want to keep on using fields?

**Comment by stephenhillier** [comment]
[EPIC#Submission | Well Decommission - Decommission Description |5| !!! [BV: 35]](https://trello.com/c/Trd9bsdp/1221-epicsubmission-well-decommission-decommission-description-5-bv-35)

**Pull request merged**